device/globals/game.gd

```diff
@@ -26,14 +26,44 @@ var tooltip
 func set_mode(p_mode):
 	mode = p_mode
 
+func tooltip_check_clamp(tt_pos):
+	if !tooltip:
+		return
+
+	var vp_size = get_viewport().size
+	var tt_size = tooltip.get_size()
+	tt_pos -= tt_size / Vector2(2, 1)
+
+	# var dist_from_right = vp_size.x - (tt_pos.x + tt_size.x)
+	# var dist_from_left = tt_pos.x - tt_size.x
+	var dist_from_bottom = vp_size.y - (tt_pos.y + tt_size.y)
+	var dist_from_top = tt_pos.y - tt_size.y
+
+	## XXX: Godot has serious issues with the width of the text, so tooltips need
+	## to be wide at a fixed size, which makes horizontal clamping impossible.
+	## The code is left here in case someone fixes Godot.
+	# if dist_from_right < 0:
+		# tt_pos.x += dist_from_right
+	# if dist_from_left < 0:
+		# tt_pos.x -= dist_from_left
+
+	if dist_from_bottom < 0:
+		tt_pos.y += dist_from_bottom
+	if dist_from_top < 0:
+		tt_pos.y -= dist_from_top
+
+	return tt_pos
+
 func mouse_enter(obj):
 	var text
 	var tt = obj.get_tooltip()
 
 	# When following the mouse, prevent text from flashing for a moment in the wrong place
 	if ProjectSettings.get_setting("escoria/ui/tooltip_follows_mouse"):
 		var pos = get_viewport().get_mouse_position()
-		pos -= tooltip.get_size() / Vector2(2, 1)
+
+		pos = tooltip_check_clamp(pos)
```

Comment:
Here, I mean. Might be a tad bit easier to follow if the presence of tooltip can be checked here.

Reply:
I'll move the check upward, so it's 'if tooltip and ProjectSettings....`